src/Nethermind/Nethermind.JsonRpc.Test/Modules/BlockConstructor.cs

```diff
@@ -0,0 +1,149 @@
+            {
+                block = BlockBuilder(keyValuePair, parentHash);
+                parentHash = block.Hash;
+                blocks.Add(block);
+            }
+
+            return blocks.ToArray();
+        }
+
+        private static Block BlockBuilder(KeyValuePair<int, string[][]> keyValuePair, Keccak parentHash, bool isEip1559 = false)
+        {
+            Transaction[] transactions;
+            Block block;
+            
+            int blockNumber = keyValuePair.Key;
+            string[][] txInfoArray = keyValuePair.Value;
+            transactions = GetTransactionsFromTxStrings(txInfoArray, isEip1559);
+            block = GetBlockWithNumberParentHashAndTxInfo(blockNumber, parentHash, transactions);
+            return block;
+        }
+
+        public static Transaction[] GetTransactionsFromTxStrings(string[][] txInfo, bool isEip1559)
+        {
+            if (txInfo == null)
+            {
+                return Array.Empty<Transaction>();
+            }
+
+            if (isEip1559)
+            {
+                return ConvertToEip1559Txs(txInfo).ToArray();
+            }
+
+            return ConvertToNonEip1559Txs(txInfo).ToArray();
+        }
+
+        private static IEnumerable<Transaction> ConvertToEip1559Txs(params string[][] txsInfo)
+        {
+            PrivateKey privateKey;
+            char privateKeyLetter;
+            UInt256 gasPrice;
+            UInt256 nonce;
+            foreach (string[] txInfo in txsInfo)
+            {
+                privateKeyLetter = Convert.ToChar(txInfo[0]);
+                privateKey = PrivateKeyForLetter(privateKeyLetter);
+                gasPrice = UInt256.Parse(txInfo[1]);
+                nonce = UInt256.Parse(txInfo[2]);
+                yield return Build.A.Transaction.SignedAndResolved(privateKey).WithGasPrice(gasPrice).WithNonce(nonce)
+                    .WithType(TxType.EIP1559).TestObject;
+            }
+        }
+
+        private static Transaction[] ConvertToNonEip1559Txs(params string[][] txsInfo)
+        {
+            PrivateKey privateKey;
+            char privateKeyLetter;
+            UInt256 gasPrice;
+            UInt256 nonce;
+            Transaction transaction;
+            List<Transaction> transactions = new List<Transaction>();
+            foreach (string[] txInfo in txsInfo)
+            {
+                privateKeyLetter = Convert.ToChar(txInfo[0]);
+                privateKey = PrivateKeyForLetter(privateKeyLetter);
+                gasPrice = UInt256.Parse(txInfo[1]);
+                nonce = UInt256.Parse(txInfo[2]);
+                transaction = Build.A.Transaction.SignedAndResolved(privateKey).WithGasPrice(gasPrice).WithNonce(nonce)
+                    .TestObject;
+                transactions.Add(transaction);
+            }
+
+            return transactions.ToArray();
+        }
+
+        public static PrivateKey PrivateKeyForLetter(char privateKeyLetter)
+        {
```

Comment:
Do not do that. We can use TestItem.PrivateKeyA. There is no sense to introduce additional method